crates/sui-framework/sources/collectible/royalty.move

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// This module implements a set of basic primitives for Kiosk's
+/// Transfer Policies to improve discoverability and serve as a
+/// base for building on top.
+module sui::royalty {
+    use std::option::{Self, Option};
+    use sui::kiosk::{Self, TransferRequest, TransferPolicyCap};
+    use sui::tx_context::{sender, TxContext};
+    use sui::balance::{Self, Balance};
+    use sui::object::{Self, UID, ID};
+    use sui::coin::{Self, Coin};
+    use sui::transfer;
+    use sui::sui::SUI;
+    use sui::event;
+
+    /// Utility constant to calculate the percentage of price to require.
+    const MAX_AMOUNT: u16 = 10_000;
+
+    /// For when trying to create a new RoyaltyPolicy with more than 100%.
+    ///  Or when trying to withdraw more than stored `balance`.
+    const EIncorrectAmount: u64 = 0;
+
+    /// A transfer policy for a single type `T` which collects a certain
+    /// fee from the `kiosk` deals and stores them for the policy issuer.
+    struct RoyaltyPolicy<phantom T: key + store> has key, store {
+        id: UID,
+        /// The `TransferPolicyCap` for the `T` which is used to call
+        /// the `kiosk::allow_transfer` and allow the trade.
+        cap: TransferPolicyCap<T>,
+        /// Percentage of the trade amount which is required for the
+        /// transfer approval. Denominated in basis points.
+        /// - 10_000 = 100%
+        /// - 100 = 1%
+        /// - 1 = 0.01%
+        amount: u16,
```

Comment:
I think it's useful to have `bp` or basis points in the name here in case someone skips the docs.